driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -125,27 +132,36 @@ static String handleId(String id) {
         if (id == null)
             return null;
 
-        if (isAlphanumeric(id))
+        boolean isAlphanumericLowCase = true;
+        boolean isAlphanumeric = true;
+        for (int i = 0; i < id.length(); i++) {
+            char c = id.charAt(i);
+            if (c >= 65 && c <= 90) { // A-Z
+                isAlphanumericLowCase = false;
+            } else {
+                if (!(
+                        (c >= 48 && c <= 57) // 0-9
+                                || (c == 95) // _ (underscore)
+                                || (c >= 97 && c <= 122) // a-z
+                )) {
+                    isAlphanumeric = false;
+                    isAlphanumericLowCase = false;
+                    break;
+                }
+            }
+        }
+
+        if (isAlphanumericLowCase) {
+            return id;
+        }
```

Comment:
👍  agreed, nice improvement.
Especially since hopefully 99% of people use lower case.